main-settings/src/main/scala/sbt/unchecked.scala

```diff
@@ -0,0 +1,13 @@
+package sbt
+
+import scala.annotation.Annotation
+
+/** An annotation to designate that the annotated entity
+ *  should not be considered for additional sbt compiler checks.
+ *  These checks ensure that the DSL is predictable and prevents
+ *  users from doing dangerous things at the cost of a stricter
+ *  code structure.
+ *
+ *  @since 1.0.0
+ */
+class unchecked extends Annotation
```

Comment:
Plugin authors regularly write `import sbt._`. So I'd be surprised if there are ambiguity of "unchecked" if I tried to use the `scala.annotation.unchecked`. I think we should a new name like "lintoff" since the meaning is also slightly different.